File: programmers/로또의최고순위와최저순위.js
/*
1) 최고순위 = 이미 맞은 개수 + 0의 개수
2) 최저순위 = 이미 맞은 개수
3) 이미 맞은 개수
*/

const NUMBER = 7;
const STANDARD = 6;

function solution(lottos, win_nums) {
    let matchCnt = 0;
    let zeroCnt = 0;
    for (const lotto of lottos) {
        if (win_nums.includes(lotto)) {
            matchCnt ++;
        }
        if (lotto === 0) zeroCnt++;
    }
    return [NUMBER-(matchCnt+zeroCnt) > STANDARD ? STANDARD : NUMBER-(matchCnt+zeroCnt), NUMBER - matchCnt > STANDARD ? STANDARD : NUMBER-matchCnt];
}

File: sparta/로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    lottoCnt = 0
    zeroCnt = 0
    
    for i in range(6):
        if lottos[i] in win_nums:
            lottoCnt += 1
        if lottos[i] == 0:
            zeroCnt += 1
            
    if lottoCnt == 0 and zeroCnt == 0:
        return [6, 6]
    
    if lottoCnt == 0:
        return [7 - zeroCnt, 6]
    else:
        return [7 - (lottoCnt + zeroCnt), 7 - lottoCnt]